doc/src/manual/functions.md

```diff
@@ -328,6 +328,22 @@ Notice the extra set of parentheses in the definition of `range`.
 Without those, `range` would be a two-argument function, and this example would
 not work.
 
+## Argument evaluation order
+
+All arguments passed to a function are evaluated in lexical left-to-right order.
```

Comment:
Would be nice to show an example with keyword arguments as well.  (#23104 was where this changed IIRC.)

Reply:
This example is also very slightly misleading --- the reason `d default` gets printed after is that it actually happens in a subsequent method call. Due to the way optional arguments work, that will *also* always be lexically later, but it's technically a different reason.

Reply:
That's a good point. I tried to pare this example down from a more complicated one with even more subtle behavior involving specifying keyword arguments out of order... Is it fair to say that the rules can be summarized as: first evaluate all passed arguments from left to right wrt. the order at the call site, and then evaluate all default values from left to right wrt. the order in the method definition? This might brush some details of how default arguments work under the rug, but I don't _think_ they're relevant here.

Reply:
Yes, that's right. Really, every case is intuitive except maybe `f(a, b=1, c, d=2, ...)` where keyword and positional arguments are mixed.

Reply:
I wonder if we should call them "named arguments" for consistency with named tuples. There's precedent for that nomenclature, though both Python and Ruby opt for "keyword" instead.

https://en.wikipedia.org/wiki/Named_parameter